### PR TITLE
Reenable default cpu arch option for clang on linux.

### PR DESCRIPTION
**Pull request by toregge**
@baldersheim : please review

### CODE REVIEW

File: default_build_settings.cmake [hunk 1]
```diff
@@ -130,6 +130,8 @@ function(vespa_use_default_build_settings)
     message("-- CMAKE_SYSTEM_PROCESSOR = ${CMAKE_SYSTEM_PROCESSOR}")
     if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
       if(APPLE AND (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")))
+      elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        set(DEFAULT_VESPA_CPU_ARCH_FLAGS "-march=westmere -mtune=haswell")
```
Comment (baldersheim) [inline default_build_settings.cmake:134]: Foreslår at du sette skylake som default der. Finnes i praksis ikke noe eldre for apple.
Also add a comment why.

**Review by baldersheim — CHANGES_REQUESTED** [review]
"-march=skylake"  and a comment.

File: default_build_settings.cmake [hunk 2]
```diff
@@ -130,6 +130,8 @@ function(vespa_use_default_build_settings)
...
       else()
         if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0)
           set(DEFAULT_VESPA_CPU_ARCH_FLAGS "-march=ivybridge")
```
Comment (baldersheim) [inline default_build_settings.cmake:137]: I think you should add a comment here with reference to the gcc bugzilla ticket too.

File: default_build_settings.cmake (hunk 1 above)
Comment (toregge) [reply default_build_settings.cmake:135]: Added comment.

File: default_build_settings.cmake (hunk 2 above)
Comment (toregge) [reply default_build_settings.cmake:139]: Added comment.

**Review by baldersheim — APPROVED** [review]

**Pull request merged**